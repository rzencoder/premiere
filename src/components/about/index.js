import React from "react";
import { Container, Item, Divider, Text, Title, Icon } from "./styles/about";

export default function About({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

About.Item = function AboutItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

About.Divider = function AboutDivider({ children, ...restProps }) {
  return <Divider {...restProps} />;
};

About.Title = function AboutTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

About.Text = function AboutText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

About.Icon = function AboutIcon({ src, ...restProps }) {
  return <Icon {...restProps} src={src} />;
};
