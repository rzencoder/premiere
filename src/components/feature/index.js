import React from "react";
import {
  Background,
  Container,
  Overlay,
  TextContainer,
  Title,
  Text,
  Button,
  Details,
  SubText,
  Rating,
  AgeRating,
  Year,
} from "./styles/feature";

export default function Feature({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Feature.Background = function FeatureBackground({ ...restProps }) {
  return <Background {...restProps} />;
};

Feature.Overlay = function FeatureOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Feature.TextContainer = function FeatureTextContainer({
  children,
  ...restProps
}) {
  return <TextContainer {...restProps}>{children}</TextContainer>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Text = function FeatureText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Feature.Button = function FeatureButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Feature.Details = function FeatureDetails({ children, ...restProps }) {
  return <Details {...restProps}>{children}</Details>;
};

Feature.SubText = function FeatureSubText({ children, ...restProps }) {
  return <SubText {...restProps}>{children}</SubText>;
};

Feature.AgeRating = function FeatureAgeRating({ ...restProps }) {
  return <AgeRating {...restProps} />;
};

Feature.Year = function FeatureYear({ children, ...restProps }) {
  return (
    <Year {...restProps}>
      <div>{children}</div>
    </Year>
  );
};

Feature.Rating = function FeatureRating({ children, ...restProps }) {
  return (
    <Rating {...restProps}>
      <div>{children}</div>
    </Rating>
  );
};
