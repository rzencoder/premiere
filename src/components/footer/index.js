import React from "react";
import {
  Container,
  Text,
  Subtitle,
  Section,
  Link,
  Divider,
  Column,
  Break,
} from "./styles/footer";

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Footer.Subtitle = function FooterSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Divider = function FooterDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Footer.Section = function FooterSection({ children, ...restProps }) {
  return <Section {...restProps}>{children}</Section>;
};

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Break = function FooterBreak({ ...restProps }) {
  return <Break {...restProps} />;
};
