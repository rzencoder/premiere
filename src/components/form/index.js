import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Container,
  Title,
  Content,
  Label,
  Input,
  Submit,
  Frame,
  Text,
  Link,
  Error,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Frame = function FormFrame({ children, ...restProps }) {
  return (
    <Frame {...restProps}>
      <div>{children}</div>
    </Frame>
  );
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Content = function FormContent({ children, ...restProps }) {
  return <Content {...restProps}>{children}</Content>;
};

Form.Label = function FormLabel({ children, ...restProps }) {
  return <Label {...restProps}>{children}</Label>;
};

Form.Input = function FormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.Link = function FormLink({ to, children, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Link {...restProps}>{children}</Link>
    </ReactRouterLink>
  );
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};
