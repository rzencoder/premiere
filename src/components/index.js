export { default as Navigation } from "./navigation";
export { default as Jumbotron } from "./jumbotron";
export { default as About } from "./about";
export { default as Footer } from "./footer";
export { default as Form } from "./form";
