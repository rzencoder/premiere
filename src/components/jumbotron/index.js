import React from "react";
import {
  Background,
  Container,
  Overlay,
  Group,
  Title,
  Text,
  CallToAction,
} from "./styles/jumbotron";
import { Link as ReactRouterLink } from "react-router-dom";

export default function Jumbotron({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Jumbotron.Overlay = function JumbotronOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Jumbotron.Group = function JumbotronGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.Text = function JumbotronText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Jumbotron.CallToAction = function JumbotronCallToAction({
  to,
  children,
  ...restProps
}) {
  return (
    <ReactRouterLink to={to}>
      <CallToAction {...restProps}>{children}</CallToAction>
    </ReactRouterLink>
  );
};

Jumbotron.Background = function JumbotronBackground({ ...restProps }) {
  return <Background {...restProps} />;
};
