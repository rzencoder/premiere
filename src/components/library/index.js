import React from "react";
import {
  Container,
  Image,
  Button,
  Divider,
  CarouselContainer,
  CarouselGroup,
  CarouselButton,
  Selector,
  SelectorItem,
  CarouselOverlay,
  Panel,
} from "./styles/library";

export default function Library({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Library.Button = function LibraryButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Library.Divider = function LibraryDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Library.Image = function LibraryImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Library.CarouselContainer = function LibraryCarouselContainer({
  children,
  ...restProps
}) {
  return <CarouselContainer {...restProps}>{children}</CarouselContainer>;
};

Library.CarouselGroup = function LibraryCarouselGroup({
  children,
  ...restProps
}) {
  return <CarouselGroup {...restProps}>{children}</CarouselGroup>;
};

Library.CarouselButton = function LibraryCarouselButton({
  children,
  ...restProps
}) {
  return <CarouselButton {...restProps}>{children}</CarouselButton>;
};

Library.Selector = function LibrarySelector({ children, ...restProps }) {
  return (
    <Selector {...restProps}>
      <div>{children}</div>
    </Selector>
  );
};

Library.SelectorItem = function LibrarySelectorItem({
  children,
  ...restProps
}) {
  return <SelectorItem {...restProps}>{children}</SelectorItem>;
};

Library.CarouselOverlay = function LibraryCarouselOverlay({ ...restProps }) {
  return <CarouselOverlay {...restProps} />;
};

Library.Panel = function LibraryPanel({ children, ...restProps }) {
  return (
    <Panel {...restProps}>
      <div>{children}</div>
    </Panel>
  );
};
