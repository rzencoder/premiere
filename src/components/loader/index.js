import React from "react";
import { Container, Spinner } from "./styles/loader";

export default function Loader({ ...restProps }) {
  return (
    <Container data-testid="loader">
      <Spinner {...restProps} />;
    </Container>
  );
}
