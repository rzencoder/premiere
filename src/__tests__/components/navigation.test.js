import React from "react";
import { render } from "@testing-library/react";
import { Navigation } from "../../components";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

describe("<Note />", () => {
  const history = createMemoryHistory();

  it("renders <Navigation /> component with home page data", () => {
    const { container, getByText, getByAltText } = render(
      <Router history={history}>
        <Navigation>
          <Navigation.Logo src="/images/logo.png" alt="premiere" to="/" />
          <Navigation.Link to="/signin">
            <Navigation.Button>Sign In</Navigation.Button>
          </Navigation.Link>
        </Navigation>
      </Router>
    );

    expect(getByText("Sign In")).toBeTruthy();
    expect(getByAltText("premiere")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("render <Navigation/> with feature page data", () => {
    const { container, getByText, getByAltText } = render(
      <Navigation>
        <Navigation.Group>
          <Navigation.BrowseLogo src="images/logo.png" alt="Premiere" />
          <Navigation.Section>
            <Navigation.Item active={true} onClick={() => {}}>
              Films
            </Navigation.Item>
            <Navigation.Item active={false} onClick={() => {}}>
              Series
            </Navigation.Item>
          </Navigation.Section>
        </Navigation.Group>
        <Navigation.Group>
          <Navigation.Search>
            <Navigation.SearchBox open={true} onChange={() => {}} id="search" />
            <Navigation.SearchIcon
              onClick={() => {}}
              tabIndex={0}
              onKeyDown={() => {}}
              src="/images/icons/search.png"
              alt="Search"
            />
          </Navigation.Search>
        </Navigation.Group>
      </Navigation>
    );

    expect(getByText("Films")).toBeTruthy();
    expect(getByAltText("Premiere")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("search style to change when closed", () => {
    const { container, getByAltText } = render(
      <Navigation.Search>
        <Navigation.SearchBox open={false} onChange={() => {}} id="search" />
        <Navigation.SearchIcon
          onClick={() => {}}
          tabIndex={0}
          onKeyDown={() => {}}
          src="/images/icons/search.png"
          alt="Search"
        />
      </Navigation.Search>
    );

    expect(getByAltText("Search")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
