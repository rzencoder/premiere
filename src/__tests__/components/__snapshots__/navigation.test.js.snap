// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> render <Navigation/> with feature page data 1`] = `
<nav
  class="navigation__Container-sc-9423gh-0 ebLQUS"
>
  <div
    class="navigation__Group-sc-9423gh-9 AOqEL"
  >
    <div
      class="navigation__BrowseLogo-sc-9423gh-5 cumtKo"
    >
      <img
        alt="Premiere"
        src="images/logo.png"
      />
    </div>
    <div
      class="navigation__Section-sc-9423gh-3 jblhpe"
    >
      <button
        class="navigation__Item-sc-9423gh-4 cmwxa-D"
      >
        Films
      </button>
      <button
        class="navigation__Item-sc-9423gh-4 dzZMMt"
      >
        Series
      </button>
    </div>
  </div>
  <div
    class="navigation__Group-sc-9423gh-9 AOqEL"
  >
    <div
      class="navigation__Search-sc-9423gh-6 ecWLfQ"
    >
      <input
        class="navigation__SearchBox-sc-9423gh-8 kUDdUz"
        id="search"
        open=""
      />
      <img
        alt="Search"
        class="navigation__SearchIcon-sc-9423gh-7 kcjsSD"
        src="/images/icons/search.png"
        tabindex="0"
      />
    </div>
  </div>
</nav>
`;

exports[`<Note /> renders <Navigation /> component with home page data 1`] = `
<nav
  class="navigation__Container-sc-9423gh-0 ebLQUS"
>
  <a
    href="/"
  >
    <img
      alt="premiere"
      class="navigation__Logo-sc-9423gh-2 hzGpvq"
      src="/images/logo.png"
    />
  </a>
  <a
    href="/signin"
  >
    <button
      class="navigation__Button-sc-9423gh-1 kaGUiX"
    >
      Sign In
    </button>
  </a>
</nav>
`;

exports[`<Note /> search style to change when closed 1`] = `
<div
  class="navigation__Search-sc-9423gh-6 ecWLfQ"
>
  <input
    class="navigation__SearchBox-sc-9423gh-8 egrHlf"
    id="search"
  />
  <img
    alt="Search"
    class="navigation__SearchIcon-sc-9423gh-7 kcjsSD"
    src="/images/icons/search.png"
    tabindex="0"
  />
</div>
`;
